src/main/java/com/smartsheet/api/internal/ReportResourcesImpl.java

```diff
@@ -0,0 +1,251 @@
+package com.smartsheet.api.internal;
+
+/*
+ * #[license]
+ * Smartsheet SDK for Java
+ * %%
+ * Copyright (C) 2014 Smartsheet
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * %[license]
+ */
+
+
+import com.smartsheet.api.ReportResources;
+import com.smartsheet.api.Smartsheet;
+import com.smartsheet.api.SmartsheetException;
+import com.smartsheet.api.internal.util.QueryUtil;
+import com.smartsheet.api.models.*;
+import java.io.OutputStream;
+import java.util.EnumSet;
+import java.util.HashMap;
+
+/**
+ * This is the implementation of the ReportResources.
+ *
+ * Thread Safety: This class is thread safe because it is immutable and its base class is thread safe.
+ */
+
+public class ReportResourcesImpl extends AbstractResources implements ReportResources{
+
+    /**
+     * Constructor.
+     *
+     * Parameters: - smartsheet : the SmartsheetImpl
+     *
+     * Exceptions: - IllegalArgumentException : if any argument is null
+     *
+     * @param smartsheet the smartsheet
+     */
+    public ReportResourcesImpl(SmartsheetImpl smartsheet) {
+        super(smartsheet);
+    }
+
+    /**
+     * Get a report.
+     *
+     * It mirrors to the following Smartsheet REST API method: GET /reports/{id}
+     *
+     * Exceptions:
+     *   InvalidRequestException : if there is any problem with the REST API request
+     *   AuthorizationException : if there is any problem with the REST API authorization(access token)
+     *   ResourceNotFoundException : if the resource can not be found
+     *   ServiceUnavailableException : if the REST API service is not available (possibly due to rate limiting)
+     *   SmartsheetRestException : if there is any other REST API related error occurred during the operation
+     *   SmartsheetException : if there is any other error occurred during the operation
+     *
+     * @param reportId the folder id
+     * @param includes the optional objects to include in response
+     * @param pageSize Number of rows per page
+     * @param page page number to return
+     * @return the report (note that if there is no such resource, this method will throw ResourceNotFoundException
+     * rather than returning null)
+     * @throws SmartsheetException the smartsheet exception
+     */
+    public Report getReport(long reportId, EnumSet<ReportInclusion> includes, Integer pageSize, Integer page) throws SmartsheetException{
+        this.checkParameters(pageSize, page);
```

Comment:
Let's not use the checkParameters() validation for two reasons:
1. We are already checking for null in this function
2. We are preventing inputs that the API allows. Instead we should let the API return an error if the input is not valid since it already does this. By letting the API return the error we won't find ourselves trying to recreate all of the validation that the API already does or constantly changing our validation as the API changes theirs.